File: example01/command/StatusCommand.java
package boinsoft.example01.command;

import boinsoft.example01.*;
import org.osgi.service.component.annotations.*;

@Component(
    property = {"osgi.command.scope=provision", "osgi.command.function=status"},
    immediate = true,
    service = Object.class)
public class StatusCommand {

  @Reference private ProvisionService provisionService_;

  public String status(Handle h) {
    return provisionService_.checkStatus(h).message();
  }
}

File: example02/command/StatusCommand.kt
package boinsoft.example02.command;

import boinsoft.example02.*;
import org.osgi.service.component.annotations.*;

@Component(
    property = ["osgi.command.scope=example02", "osgi.command.function=status"],
    immediate = true,
    service = [Object::class])
class StatusCommand {

  @Reference
  var provisionService_: ProvisionService? = null;

  fun status(h: Handle): String {
    var x = provisionService_?.checkStatus(h)
    if (x != null) { return x.message() }
    else return ""
  }
}
